pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/PhysicianInstitutionComponent.java

```diff
@@ -10,66 +18,106 @@
-        parameters.put(REQUIRED, required);
+    private final void setSerializedFields() {
+        serializedFields.setAllowMultiple(instDto.getAllowMultiple());
+        serializedFields.setButtonText(buttonText);
+        serializedFields.setTitleText(titleText);
+        serializedFields.setSubtitleText(subtitleText);
+        serializedFields.setInstitutionType(instDto.getInstitutionType().name());
+        serializedFields.setShowFields(instDto.showFields());
+        serializedFields.setIsRequired(instDto.isRequired());
+    }
+
+    @Override
+    public void registerTemplateIds(Consumer<Long> registry) {
+        super.registerTemplateIds(registry);
+        Optional.ofNullable(instDto.getButtonTemplateId()).ifPresentOrElse(registry::accept, () -> { });
+        Optional.ofNullable(instDto.getTitleTemplateId()).ifPresentOrElse(registry::accept, () -> { });
+        Optional.ofNullable(instDto.getSubtitleTemplateId()).ifPresentOrElse(registry::accept, () -> { });
+    }
+
+    @Override
+    public void applyRenderedTemplates(Provider<String> rendered, ContentStyle style) {
+        super.applyRenderedTemplates(rendered, style);
+        buttonText = rendered.get(instDto.getButtonTemplateId());
+        titleText = rendered.get(instDto.getTitleTemplateId());
+        subtitleText = rendered.get(instDto.getSubtitleTemplateId());
+
+        setSerializedFields();
     }
 
     public InstitutionType getInstitutionType() {
-        return institutionType;
+        return instDto.getInstitutionType();
+    }
+
+    public static class SerializedFields {
+        @SerializedName("allowMultiple")
+        private boolean allowMultiple;
+        @SerializedName("addButtonText")
+        private String buttonText;
+        @SerializedName("titleText")
+        private String titleText;
+        @SerializedName("subtitleText")
+        private String subtitleText;
+        @SerializedName("institutionType")
+        private String institutionType;
+        @SerializedName("showFieldsInitially")
+        private boolean showFields;
+        @SerializedName("required")
+        private boolean isRequired;
+
+        public void setAllowMultiple(boolean allowMultiple) {
+            this.allowMultiple = allowMultiple;
+        }
+
+        public void setButtonText(String buttonText) {
+            this.buttonText = buttonText;
+        }
+
+        public void setTitleText(String titleText) {
+            this.titleText = titleText;
+        }
+
+        public void setSubtitleText(String subtitleText) {
+            this.subtitleText = subtitleText = subtitleText;
+        }
+
+        public void setInstitutionType(String institutionType) {
+            this.institutionType = institutionType;
+        }
+
+        public void setShowFields(boolean showFields) {
+            this.showFields = showFields;
+        }
+
+        public void setIsRequired(boolean isRequired) {
+            this.isRequired = isRequired;
+        }
```

Comment:
This class replaces the generic `Map<String, Object>` in `FormComponent`.

Reply:
Nice!